modules/migrate/application/clicommands/ToicingadbCommand.php

```diff
@@ -0,0 +1,584 @@
+        $roleName = $this->params->get('role');
+
+        if ($roleName === null && $groupName === null) {
+            $this->fail("One of the parameters 'group' or 'role' must be supplied");
+        }
+
+        $rc = 0;
+        $restrictions = Config::$configDir . '/roles.ini';
+        $rolesConfig = $this->readFromIni($restrictions, $rc);
+        $monitoringRestriction = 'monitoring/filter/objects';
+        $monitoringPropertyBlackList = 'monitoring/blacklist/properties';
+        $icingadbRestrictions = [
+            'objects'  => 'icingadb/filter/objects',
+            'hosts'    => 'icingadb/filter/hosts',
+            'services' => 'icingadb/filter/services'
+        ];
+
+        $icingadbPropertyDenyList = 'icingadb/denylist/variables';
+        Logger::info('Start monitoring restrictions migration');
+        foreach ($rolesConfig as $name => $role) {
+            /** @var string[] $role */
+            $role = iterator_to_array($role);
+            $updateRole = false;
+
+            if ($roleName === '*' || $groupName === '*') {
+                $updateRole = true;
+            } elseif ($roleName !== null && fnmatch($roleName, $name)) {
+                $updateRole = true;
+            } elseif ($groupName !== null && isset($role['groups'])) {
+                $roleGroups = array_map('trim', explode(',', $role['groups']));
+
+                foreach ($roleGroups as $roleGroup) {
+                    if (fnmatch($groupName, $roleGroup)) {
+                        $updateRole = true;
+                        break;
+                    }
+                }
+            }
+
+            if ($updateRole) {
+                if (isset($role[$monitoringRestriction])) {
+                    if (! isset($role[$icingadbRestrictions['objects']]) || $override) {
+                        Logger::info(
+                            'Migrating monitoring restriction filter for role "%s" to the Icinga DB Web restrictions',
+                            $name
+                        );
+                        $transformedFilter = UrlMigrator::transformFilter(
+                            QueryString::parse($role[$monitoringRestriction])
+                        );
+
+                        if ($transformedFilter) {
+                            $role[$icingadbRestrictions['objects']] = rawurldecode(
+                                QueryString::render($transformedFilter)
+                            );
+                        }
+                    }
+                }
+
+                if (isset($role[$monitoringPropertyBlackList])) {
+                    if (! isset($role[$icingadbPropertyDenyList]) || $override) {
+                        Logger::info(
+                            'Migrating monitoring blacklisted properties for role "%s" to the Icinga DB Web deny list',
+                            $name
+                        );
+                        $role[$icingadbPropertyDenyList] = $role[$monitoringPropertyBlackList];
```

Comment:
these can't just be copied over. See https://icinga.com/docs/icinga-db-web/latest/doc/10-Migration/#monitoringblacklistproperties and https://icinga.com/docs/icinga-web/latest/modules/monitoring/doc/10-Restrict-Custom-Variables/